Sources/StreamChat/Database/DTOs/ChannelConfigDTO.swift

```diff
@@ -0,0 +1,76 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import CoreData
+import Foundation
+
+@objc(ChannelConfigDTO)
+class ChannelConfigDTO: NSManagedObject {
+    @NSManaged var reactionsEnabled: Bool
+    @NSManaged var typingEventsEnabled: Bool
+    @NSManaged var readEventsEnabled: Bool
+    @NSManaged var connectEventsEnabled: Bool
+    @NSManaged var uploadsEnabled: Bool
+    @NSManaged var repliesEnabled: Bool
+    @NSManaged var quotesEnabled: Bool
+    @NSManaged var searchEnabled: Bool
+    @NSManaged var mutesEnabled: Bool
+    @NSManaged var urlEnrichmentEnabled: Bool
+    @NSManaged var messageRetention: String
+    @NSManaged var maxMessageLength: Int32
+    @NSManaged var createdAt: Date
+    @NSManaged var updatedAt: Date
+    @NSManaged var commands: Set<CommandDTO>
+
+    func asModel() -> ChannelConfig {
+        .init(
+            reactionsEnabled: reactionsEnabled,
+            typingEventsEnabled: typingEventsEnabled,
+            readEventsEnabled: readEventsEnabled,
+            connectEventsEnabled: connectEventsEnabled,
+            uploadsEnabled: uploadsEnabled,
+            repliesEnabled: repliesEnabled,
+            quotesEnabled: quotesEnabled,
+            searchEnabled: searchEnabled,
+            mutesEnabled: mutesEnabled,
+            urlEnrichmentEnabled: urlEnrichmentEnabled,
+            messageRetention: messageRetention,
+            maxMessageLength: Int(maxMessageLength),
+            commands: commands.map { $0.asModel() },
+            createdAt: createdAt,
+            updatedAt: updatedAt
+        )
+    }
+}
+
+extension ChannelConfig {
```

Comment:
Since the PR is focusing on `ChannelConfig`, can we clean it up a bit? Make it `Decodable` instead of `Codable` (as we no longer encode it), make it a struct to be aligned with other models/payloads, remove manually written initializer and leverage the generated one?

Reply:
This was changed to a `class` to reduce the size of the SDK -> https://github.com/GetStream/stream-chat-swift/pull/1549

The Codable to Decodable, and the manual initializer makes sense 👍

Reply:
Actually, @evsaev I'm not going to do this change. Changing from `Codable -> Decodable` is a breaking change. If a customer is using for example a JSONEncoder to mock the ChannelConfig, this will stop working. Since Codable API will be public as well in this case.

Reply:
The root cause of this issue is that we are missing a data structure to define the config from the Web API. Right now our domain modal is also the web model as well

Reply:
Yeah, ideally there should always be a payload/model pair. The same for command. Can we create a chore task for that?

> Actually, @evsaev I'm not going to do this change. Changing from Codable -> Decodable is a breaking change. If a customer is using for example a JSONEncoder to mock the ChannelConfig, this will stop working. Codable API is public

Let's add this to v5 wish list so we don't forget to fix it 👍